src/ast/irbuilderbpf.cpp

```diff
@@ -200,6 +209,54 @@ AllocaInst *IRBuilderBPF::CreateAllocaBPF(int bytes, const std::string &name)
   return CreateAllocaBPF(ty, name);
 }
 
+void IRBuilderBPF::CreateMemsetBPF(Value *ctx,
+                                   Value *ptr,
+                                   Value *val,
+                                   uint32_t size)
+{
+  assert(ctx && ctx->getType() == GET_PTR_TY());
```

Comment:
Looks like ctx is unused in this function beside the assertion. That on purpose?

Reply:
Looks like it's the reason you had to pass ctx around for all these functions.

Reply:
Ah good catch. Previously I was calling the CreateProbeRead() function which required a ctx. But eventually I realized it wasn't suitable for this use case. Forgot to delete ctx. Will fix.